File: riveredge-frontend/src/services/sensitiveWordBlacklist.ts
import { apiRequest } from './api';

export interface SensitiveWordBlacklistMeta {
  menu_visible: boolean;
  enabled_tenant_count: number;
  enabled_tenants: Array<{ id: number; name: string; domain: string }>;
}

export interface SensitiveWordBanItem {
  id: number;
  tenant_id: number;
  tenant_name?: string | null;
  user_id: number;
  username?: string | null;
  full_name?: string | null;
  client_ip: string;
  banned_at: string;
  unbanned_at?: string | null;
  is_active: boolean;
  trigger_request_path?: string | null;
  trigger_field_path?: string | null;
  trigger_matched_word?: string | null;
  trigger_content_snippet?: string | null;
}

export interface SensitiveWordBanListResponse {
  items: SensitiveWordBanItem[];
  total: number;
  page: number;
  page_size: number;
}

export interface TenantSensitiveWordAllowlistItem {
  id: number;
  tenant_id: number;
  word: string;
  note?: string | null;
  created_at: string;
}

export interface TenantSensitiveWordAllowlistListResponse {
  items: TenantSensitiveWordAllowlistItem[];
  total: number;
  page: number;
  page_size: number;
}

export async function getSensitiveWordBlacklistMeta(): Promise<SensitiveWordBlacklistMeta> {
  return apiRequest<SensitiveWordBlacklistMeta>('/infra/sensitive-word-blacklist/meta');
}

export async function listSensitiveWordBans(params: {
  page?: number;
  page_size?: number;
  tenant_id?: number;
  active_only?: boolean;
}): Promise<SensitiveWordBanListResponse> {
  return apiRequest<SensitiveWordBanListResponse>('/infra/sensitive-word-blacklist/bans', {
    params,
  });
}

export async function unbanSensitiveWordSubject(banId: number): Promise<void> {
  await apiRequest(`/infra/sensitive-word-blacklist/bans/${banId}/unban`, {
    method: 'POST',
  });
}

export async function listTenantSensitiveWordAllowlist(params: {
  tenant_id: number;
  page?: number;
  page_size?: number;
}): Promise<TenantSensitiveWordAllowlistListResponse> {
  return apiRequest<TenantSensitiveWordAllowlistListResponse>('/infra/sensitive-word-blacklist/allowlist', {
    params,
  });
}

export async function addTenantSensitiveWordAllowlist(data: {
  tenant_id: number;
  word: string;
  note?: string;
}): Promise<TenantSensitiveWordAllowlistItem> {
  return apiRequest<TenantSensitiveWordAllowlistItem>('/infra/sensitive-word-blacklist/allowlist', {
    method: 'POST',
    data,
  });
}

export async function removeTenantSensitiveWordAllowlist(allowlistId: number): Promise<void> {
  await apiRequest(`/infra/sensitive-word-blacklist/allowlist/${allowlistId}`, {
    method: 'DELETE',
  });
}

File: riveredge-backend/src/infra/schemas/sensitive_word_blacklist.py
"""敏感词黑名单 API 模型。"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class SensitiveWordBlacklistMetaResponse(BaseModel):
    menu_visible: bool
    enabled_tenant_count: int
    enabled_tenants: List[dict] = Field(default_factory=list)


class SensitiveWordBanItem(BaseModel):
    id: int
    tenant_id: int
    tenant_name: Optional[str] = None
    user_id: int
    username: Optional[str] = None
    full_name: Optional[str] = None
    client_ip: str
    banned_at: datetime
    unbanned_at: Optional[datetime] = None
    is_active: bool
    trigger_request_path: Optional[str] = None
    trigger_field_path: Optional[str] = None
    trigger_matched_word: Optional[str] = None
    trigger_content_snippet: Optional[str] = None


class SensitiveWordBanListResponse(BaseModel):
    items: List[SensitiveWordBanItem]
    total: int
    page: int
    page_size: int


class TenantSensitiveWordAllowlistItem(BaseModel):
    id: int
    tenant_id: int
    word: str
    note: Optional[str] = None
    created_at: datetime


class TenantSensitiveWordAllowlistListResponse(BaseModel):
    items: List[TenantSensitiveWordAllowlistItem]
    total: int
    page: int
    page_size: int


class TenantSensitiveWordAllowlistCreate(BaseModel):
    tenant_id: int = Field(..., description="组织 ID")
    word: str = Field(..., min_length=1, max_length=128)
    note: Optional[str] = Field(None, max_length=255)
